File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left_pointer = 0
        right_pointer = len(numbers) - 1
        i = 0
        answer_list = list()


        while(left_pointer < right_pointer):
            if(numbers[left_pointer] + numbers[right_pointer] == target):
                answer_list.append(left_pointer + 1)
                answer_list.append(right_pointer + 1)
                return answer_list
            elif(numbers[left_pointer] + numbers[right_pointer] > target):
                right_pointer = right_pointer - 1

            else:
                left_pointer = left_pointer + 1
        
        return None
            

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
    	
    	int pointer_left = 0;
    	int pointer_right = numbers.length-1;
        int[] answer_array = new int[2];
        
        while(pointer_left < pointer_right) {
            	
	    	if(numbers[pointer_left] + numbers[pointer_right] == target) {
	    		answer_array[0] = pointer_left+1;
	    		answer_array[1] = pointer_right+1;
                return answer_array;
	    	}
	    	else if(numbers[pointer_left] + numbers[pointer_right] > target) {
	    		pointer_right--;
	    	}
	    	else {
	    		pointer_left++;
	    	}
        }
        return null;
    }
}